src/utils/merge.js

```diff
@@ -19,9 +22,40 @@ export default function mergeObjects (dest, ...sources) {
                 }
             }
             // Overwrite the previous destination value if the source property is: a scalar (number/string),
-            // an array, or a null value
+            // an array (unless using array merge operator below), or a null value
             else if (value !== undefined) {
-                dest[key] = value;
+                let merge_index = is_array && value.indexOf(array_merge_op); // special array merge syntax
+                if (is_array && dest[key] != null && merge_index > -1) {
+                    // If the source value includes the '...' operator, the previous destination value will be
+                    // merged into the source value in its place.
+                    //
+                    // Example 1: append values
+                    //   dest: [1, 2 3]
+                    //   source: ['...', 4, 5, 6]
+                    //   -> merge result: [1, 2, 3, 4, 5, 6]
+                    //
+                    // Example 2: insert in middle
+                    //   dest: [1, 2 3]
+                    //   source: ['a', '...', 'b']
+                    //   -> merge result: ['a', 1, 2, 3, 'b']
+                    //
+                    // Example 3: existing scalar value in destination is treated as a single-element array
+                    //   dest: 'x'
+                    //   source: [1, 2, 3, '...']
+                    //   -> merge result: [1, 2, 3, 'x']
+                    //
+                    if (!Array.isArray(dest[key])) {
```

Comment:
The comments describe this behavior for "scalar" values but this would appear to treat a mapping/object as a "single-element array" as well. Is this intended?